src/main/java/com/microsoft/graph/requests/middleware/GraphTelemetryHandler.java

```diff
@@ -43,16 +44,16 @@ public Response intercept(@Nonnull final Chain chain) throws IOException {
         final Request request = chain.request();
         final Request.Builder telemetryAddedBuilder = request.newBuilder();
 
-        final String graphEndpoint = mGraphClientOption.getGraphServiceTargetVersion();
-        final String featureUsage = "(featureUsage=" + mGraphClientOption.featureTracker.getSerializedFeatureUsage() + ")";
+        final String clientLibraryUsed = "graph-java" + (mGraphClientOption.getGraphServiceTargetVersion().equals("beta") ? "-beta" : "");
```

Comment:
Rather than hardcoding the check for beta, maybe we should just check if the `mGraphClientOption.getGraphServiceTargetVersion()` string is not empty and then append `-version`. Then in the mGraphClientOption can be set to an empty string when we don't want to pass a version here. (This will avoid issues if we possibly have a library that is not beta)

Reply:
We did that originally, changed it in this pr simply because we don't want to append -v1.0 since we set the graphServiceTargetVersion to v1.0 in the [service client](https://github.com/microsoftgraph/msgraph-sdk-java/blob/284ea18198cc8d0e9fe89ab8f22685f16feb9b5c/src/main/java/com/microsoft/serviceClient/GraphServiceClient.java#LL24C4-L29C6). You're right though, it shouldn't be hardcoded. Will add a check, if graphServiceClient is not 'v1.0' then append it, otherwise leave it as is.